lib/JSON/Validator.pm

```diff
@@ -719,8 +719,12 @@ sub _guessed_right {
 }
 
 sub _is_date_time {
-  $_[0] =~ qr/^(\d+)-(\d+)-(\d+)T(\d+):(\d+):(\d+(?:\.\d+)?)(?:Z|([+-])(\d+):(\d+))?$/io;
+  $_[0] =~ qr/
+    ^(\d+)-(\d+)-(\d+)\D+(\d+):(\d+):(\d+(?:\.\d+)?)   # Date and time
```

Comment:
Two things:

* While you're making this change, could you remove all the captures? There's no need for it here, since it's just a validator.
* I'm pretty sure `\D+` is incorrect. If it indeed allows any character to be used as separator it should just be `.`.